src/ManageCourses.Api/Middleware/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.AspNetCore.Http;
+
+namespace GovUk.Education.ManageCourses.Api.Middleware
+{
+    public static class HttpRequestExtensions
+    {
+        public static string GetAccessToken(this HttpRequest request)
+        {
+            var authorization = "Authorization";
+            var authorizationHeaderValues = request.Headers.ContainsKey(authorization) ?
+                ((string)request.Headers[authorization]).Split(' ') : new[] { "", "" };
+
+            var accessToken = authorizationHeaderValues.Length == 2 && authorizationHeaderValues[0].ToLowerInvariant().Equals("bearer") ?
+                authorizationHeaderValues[1] : "";
```

Comment:
I think this would be hard to troubleshoot if any of the assumptions about the contents of the header failed.
I would throw if the assumptions are incorrect.

Reply:
There is no assumption.

Headers was checked to see if it contains key auth.
auth value is split to array or a default of array with 2 empty string
if array is 2 return second value and if the first equals `bearer`

negative results will always be empty string 

throwing exception is expensive

if it contained auth  key but a null value then that is where the exception should be throw as that is beyond expectation

Reply:
The assumption I noticed was that the split would always result in two parts, and silently fail if that was not the case. Maybe that's never going to change but looking at the method in isolation it struck me as a smell that it was just swallowed. I wouldn't reject the pr for this though.